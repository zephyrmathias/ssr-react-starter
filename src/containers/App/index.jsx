import React from 'react';
import { renderRoutes } from 'react-router-config';
import PropTypes from 'prop-types';
import { hot } from 'react-hot-loader';
import Header from '../../components/Header';

const App = ({ route }) => (
  <div>
    <Header />
    <div>
      App
    </div>
    {renderRoutes(route.routes)}
  </div>
);

App.propTypes = {
  route: PropTypes.shape({
    component: PropTypes.func.isRequired,
    routes: PropTypes.arrayOf(
      PropTypes.shape({
        component: PropTypes.func.isRequired,
        path: PropTypes.string,
        exact: PropTypes.bool,
      }),
    ).isRequired,
  }).isRequired,
};


export default hot(module)(App);
