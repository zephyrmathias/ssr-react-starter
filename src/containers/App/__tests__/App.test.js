import React from 'react';
import TestRenderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import App from '../index';

describe('containers: <App />', () => {
  test('should render <App /> with routes properly', () => {
    const fakeRoutes = {
      component: () => (
        <div>
          fake component
        </div>
      ),
      routes: [
        {
          path: '/',
          exact: true,
          component: () => (
            <div>
              fake component
            </div>
          ),
        },
      ],
    };

    const tree = TestRenderer
      .create(
        <MemoryRouter>
          <App route={fakeRoutes} />
        </MemoryRouter>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
