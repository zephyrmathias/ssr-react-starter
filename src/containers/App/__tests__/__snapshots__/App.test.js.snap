// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers: <App /> should render <App /> with routes properly 1`] = `
<div>
  <div>
    <h1>
      Header
    </h1>
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      Home
    </a>
    <a
      aria-current={null}
      href="/posts"
      onClick={[Function]}
    >
      Posts
    </a>
  </div>
  <div>
    App
  </div>
  <div>
    fake component
  </div>
</div>
`;
